src/lib.rs

```diff
@@ -827,23 +835,93 @@ impl SyncClient {
         }
     }
 
+    async fn request_wildcard_handler(
+        wrapped_client: Arc<Mutex<Self>>,
+        mut subscription_rx: MpscReceiver<Msg>,
+    ) {
+        let disconnecting = Self::disconnecting(Arc::clone(&wrapped_client));
```

Comment:
Is it possible that cleaning up at this point might cause the client to end up in a weird place if a request slips in after the cleanup in this fn runs but before the connection is officially disconnected and then reconnected?

Reply:
Is it possible to explicitly unsubscribe from `request_wildcard_subscription` after we have detected a disconnect in the [`server_messages_handler`](https://github.com/davidMcneil/rants/blob/92547d02ce65c368f6ac36ac4ce8654f1dd870c3/src/lib.rs#L1069)? This should cause `subscription_rx.next()` to return `None` which would break us out of the loop. This has two advantages. It cleans up the code a bit and ensures that we have actually disconnected.

Reply:
Great idea. I have to manually drop it from the subscriptions list since the unsubscribe method will reject everything when in a disconnecting state. But it's still a lot cleaner!